MISRA.md

```diff
@@ -1,24 +1,17 @@
 # MISRA Compliance
 
-The Device Defender Client Library files conform to the
-[MISRA C:2012](https://www.misra.org.uk)
+The Device Defender Client Library files conform to the [MISRA C:2012](https://www.misra.org.uk)
 guidelines, with some noted exceptions. Compliance is checked with Coverity static analysis.
-Deviations from the MISRA standard are listed below:
+The specific deviations, suppressed inline, are listed below.
 
-### Ignored by [Coverity Configuration](tools/coverity/misra.config)
-| Deviation | Category | Justification |
-| :-: | :-: | :-: |
-| Directive 4.9 | Advisory | Allow inclusion of function like macros. The `assert` macro is used throughout the library for parameter validation, and logging is done using function like macros. |
-| Rule 3.1 | Required | Allow nested comments. C++ style `//` comments are used in example code within Doxygen documentation blocks. |
-| Rule 12.3 | Advisory | Allow use of `assert` macro, expansion of which uses comma operator. |
-| Rule 15.6 | Required | Allow use of `assert` macro, expansion of which contains non-compound if statements. |
-| Rule 20.12 | Required | Allow use of `assert` macro, which uses a parameter in both expanded and raw forms. |
-
-### Flagged by Coverity
-| Deviation | Category | Justification |
-| :-: | :-: | :-: |
-| Rule 2.5 | Advisory | Macros defined for keys in the defender report are not used by the library; however, they are needed to be used by an application. |
-| Rule 8.7 | Advisory | API functions are not used by the library; however, they must be externally visible in order to be used by an application. |
+Additionally, [MISRA configuration file](https://github.com/aws/Device-Defender-for-AWS-IoT-embedded-sdk/blob/main/tools/coverity/misra.config) contains the project wide deviations.
 
 ### Suppressed with Coverity Comments
-*None.*
\ No newline at end of file
+To find the violation references in the source files run grep on the source code
+with ( Assuming rule 11.4 violation; with justification in point 2 ):
+```
+grep 'MISRA Ref 11.4.2' . -rI
```

Comment:
So I can agree with the idea of having line numbers/links instead of having users run grep, but I feel like that could add a lot of overhead to create a slightly better end user experience. Since then that would require every source code contribution to require also coming into this file and modifying said table. Then sometimes there will be multiple instances of a single justification being used for multiple inline suppressions, as seen here in [corePKCS11](https://github.com/FreeRTOS/corePKCS11/blob/main/source/portable/mbedtls/core_pkcs11_mbedtls.c#L397), which could lead to a single entry having a large list of line numbers. Updating this could then lead to a decent amount of developer time being used to check line numbers as part of a PR.